larpandoracontent/LArTwoDReco/LArClusterAssociation/MvaLowEClusterMergingAlgorithm.cc

```diff
@@ -0,0 +1,750 @@
+       //const MCParticle *const clusterMC(MCParticleHelper::GetMainMCParticle(cluster));
+       const int nClusters(pClusterList->size());
+       std::cout << "Number of clusters to check: " << nClusters << std::endl;	
+       
+       if (!this->IsValidToUse(cluster, clusterIsUsed))
+            continue;
+
+	CaloHitList clusterEdgeHits;
+	CartesianVector centroid11(0.f, 0.f, 0.f);
+        CartesianVector centroid12(0.f, 0.f, 0.f);
+        try {
+       	centroid11 = cluster->GetCentroid(cluster->GetInnerPseudoLayer());
+        centroid12 = cluster->GetCentroid(cluster->GetOuterPseudoLayer());
+        }
+        catch (...) {
+        CaloHitList hits;
+        cluster->GetOrderedCaloHitList().FillCaloHitList(hits);
+        centroid11 = hits.front()->GetPositionVector();
+        centroid12 = hits.front()->GetPositionVector();
+        }
+
+        //const CartesianVector centroid11(cluster->GetCentroid(cluster->GetInnerPseudoLayer()));
+        //const CartesianVector centroid12(cluster->GetCentroid(cluster->GetOuterPseudoLayer()));
+        const CartesianVector centroid1((centroid11 + centroid12) * 0.5f);
+	
+	this->EdgeHitFinder(cluster, clusterEdgeHits); 
+
+        const unsigned int clusterNHits{cluster->GetNCaloHits()};
+        const unsigned long int clusterNEdgeHits{clusterEdgeHits.size()};
+
+	std::cout << "Running 1st Loop" << std::endl;
+	for (auto iter2 = std::next(iter) ; iter2 != pClusterList->end() ; ++iter2)
+	{
+	    const Cluster *const otherCluster(*iter2);
+	    const MCParticle *otherClusterMC(this->GetMCForCluster(otherCluster, clusterToMCParticleMap));
+            sigBac = 0;
+
+            if (clusterMC == otherClusterMC)
+            {
+                   sigBac = 1;
+
+            }
+
+	    std::cout << "********* CLUSTER MC ********" << std::endl;
+            std::cout << "Cluster 1: " << clusterMC <<std::endl;
+            std::cout << "Cluster 2: " << otherClusterMC << std::endl;
+            std::cout << "Bool: " << sigBac << std::endl;    
+	    std::cout << "*****************************" << std::endl;
+
+	    CaloHitList otherClusterEdgeHits;
+	    CartesianVector centroid21(0.f, 0.f, 0.f);
+	    CartesianVector centroid22(0.f, 0.f, 0.f);
+	    try {
+	    centroid21 = otherCluster->GetCentroid(otherCluster->GetInnerPseudoLayer());
+            centroid22 = otherCluster->GetCentroid(otherCluster->GetOuterPseudoLayer());
+	    }
+	    catch (...) {
+	    CaloHitList hits;
+            cluster->GetOrderedCaloHitList().FillCaloHitList(hits);
+            centroid21 = hits.front()->GetPositionVector();
+            centroid22 = hits.front()->GetPositionVector();
+	    }
+	    const CartesianVector centroid2((centroid21 + centroid22) * 0.5f);
+	    const float centroidSeparation{Distance(centroid1, centroid2)};
+	    const CartesianVector centroidVector(centroid2.GetX() - centroid1.GetX(), 0.f, centroid2.GetZ() - centroid1.GetZ());
```

Comment:
You can use CarteisanVector's operator - to get the vector between the two centroids, and then also just get its magnitude for the separation distance.